### PR TITLE
Fix UNIX socket support for MySQL in dump action

### LINKED ISSUE #235: dump fail without setting hostname and passoword in MySQl
I just test dump function with mysql instance, that will fail with assign unix socket path.
And the code as below, there are not any args about socket ,only hostname and port.
The general command of mysqldump is  `mysqldump -uroot -S /data/mysql/run/mysql.sock`
```
func (drv *Driver) mysqldumpArgs() []string {
	// generate CLI arguments
	args := []string{"--opt", "--routines", "--no-data",
		"--skip-dump-date", "--skip-add-drop-table"}

	if hostname := drv.databaseURL.Hostname(); hostname != "" {
		args = append(args, "--host="+hostname)
	}
	if port := drv.databaseURL.Port(); port != "" {
		args = append(args, "--port="+port)
	}
	if username := drv.databaseURL.User.Username(); username != "" {
		args = append(args, "--user="+username)
	}
	// mysql recommends against using environment variables to supply password
	// https://dev.mysql.com/doc/refman/5.7/en/password-security-user.html
	if password, set := drv.databaseURL.User.Password(); set {
		args = append(args, "--password="+password)
	}

	// add database name
	args = append(args, dbutil.DatabaseName(drv.databaseURL))

	return args
}
```

### PR DESCRIPTION
**Pull request by Enrico204**
`mysqldumpArgs` was missing the `--socket` parameter when launching `mysqldump` for UNIX socket connections. This pull request adds the parameter if the `socket` query option is added in the database URL

Closes #235

### CODE REVIEW

**Comment by amacneil** [comment]
Thanks! Can you add "Closes #235" in the issue description so that it is linked to the bug and will automatically close it when merged?

File: pkg/driver/mysql/mysql.go [hunk 1]
```diff
@@ -145,6 +145,9 @@ func (drv *Driver) mysqldumpArgs() []string {
 	if password, set := drv.databaseURL.User.Password(); set {
 		args = append(args, "--password="+password)
 	}
+	if socket := drv.databaseURL.Query().Get("socket"); drv.databaseURL.Hostname() == "" && socket != "" {
```
Comment (amacneil) [inline pkg/driver/mysql/mysql.go:148]: I forget, do we have any tests for this function? Would be nice to add a test to cover the new functionality.

Comment (Enrico204) [reply pkg/driver/mysql/mysql.go:148]: Currently, no. But I might try to come up with some testing. Let me see what I can do :-)

**Comment by amacneil** [comment]
Thanks! I added some tests.

**Pull request merged**